includes/internal/emails/class-email-list-table.php

```diff
@@ -230,4 +240,108 @@ private function get_row_actions( $post ): array {
 		 */
 		return apply_filters( 'sensei_email_list_row_actions', $actions, $post, $this );
 	}
+
+	/**
+	 * Render the bulk actions.
+	 *
+	 * @param string $which The location of the bulk actions: 'top' or 'bottom'.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The bulk actions HTML.
+	 */
+	public function bulk_actions( $which = '' ) {
```

Comment:
I understand that this method exists so we can apply the design, but as Donna [said](https://github.com/Automattic/sensei/pull/6505#issuecomment-1426117989), the design is not always following WordPress but we should tend to implement things the WordPress way.

Having said that, I think we could use the built-in WordPress bulk action dropdown here and save us some maintenance time in the future. 

Here's a quick patch that explains my idea:

```diff
diff --git a/includes/class-sensei-list-table.php b/includes/class-sensei-list-table.php
index b7351b1af..4688bd3b6 100755
--- a/includes/class-sensei-list-table.php
+++ b/includes/class-sensei-list-table.php
@@ -322,8 +322,14 @@ class Sensei_List_Table extends WP_List_Table {
 	 * @since  1.2.0
 	 */
 	public function bulk_actions( $which = '' ) {
+		ob_start();
+
+		parent::bulk_actions();
+
+		$bulk_actions = ob_get_clean();
+
 		// This will be output Above the table headers on the left
-		echo wp_kses_post( apply_filters( 'sensei_list_bulk_actions', '' ) );
+		echo apply_filters( 'sensei_list_bulk_actions', $bulk_actions ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Trusted output.
 	}
 
 	/**
diff --git a/includes/internal/emails/class-email-list-table.php b/includes/internal/emails/class-email-list-table.php
index 68cb1147f..08a3d3f61 100644
--- a/includes/internal/emails/class-email-list-table.php
+++ b/includes/internal/emails/class-email-list-table.php
@@ -241,77 +241,12 @@ class Email_List_Table extends Sensei_List_Table {
 		return apply_filters( 'sensei_email_list_row_actions', $actions, $post, $this );
 	}
 
-	/**
-	 * Render the bulk actions.
-	 *
-	 * @param string $which The location of the bulk actions: 'top' or 'bottom'.
-	 *
-	 * @since $$next-version$$
-	 *
-	 * @return string The bulk actions HTML.
-	 */
-	public function bulk_actions( $which = '' ) {
-		if ( 'top' !== $which ) {
-			return '';
-		}
-
-		/**
-		 * Filter the bulk actions displayed on the email list.
-		 *
-		 * @since $$next-version$$
-		 * @hook sensei_email_list_bulk_actions
-		 *
-		 * @param {array} $bulk_actions The bulk actions.
-		 *
-		 * @return {array}
-		 */
-		$bulk_actions = apply_filters(
-			'sensei_email_list_bulk_actions',
-			[
-				'bulk-disable-email' => __( 'Disable', 'sensei-lms' ),
-				'bulk-enable-email'  => __( 'Enable', 'sensei-lms' ),
-			]
-		);
-
-		$output = '<label for="bulk-action-selector-top" class="screen-reader-text">' . __( 'Bulk Email Actions', 'sensei-lms' ) . '</label>'
-		. '<select name="action" id="bulk-action-selector-top">'
-		. '<option value="-1">' . __( 'Bulk Email Actions', 'sensei-lms' ) . '</option>';
-
-		foreach ( $bulk_actions as $key => $value ) {
-			$output .= '<option value="' . esc_attr( $key ) . '">' . $value . '</option>';
-		}
-
-		$output .= '</select>';
-
-		$output .= get_submit_button( __( 'Select Action', 'sensei-lms' ), 'action', '', false, array( 'id' => 'doaction' ) );
-
-		echo wp_kses(
-			$output,
-			[
-				'select' => [
-					'id'    => [],
-					'name'  => [],
-					'class' => [],
-				],
-				'option' => [
-					'value' => [],
-					'class' => [],
-				],
-				'input'  => [
-					'type'  => [],
-					'id'    => [],
-					'name'  => [],
-					'value' => [],
-					'class' => [],
-				],
-				'label'  => [
-					'for'   => [],
-					'class' => [],
-				],
-			]
-		);
-
-		return $output;
+	public function get_bulk_actions() {
+		// No need for a hook here. WP calls one in `WP_List_Table::bulk_actions()`.
+		return [
+			'bulk-disable-email' => __( 'Disable', 'sensei-lms' ),
+			'bulk-enable-email'  => __( 'Enable', 'sensei-lms' ),
+		];
 	}
 
 	/**
```

This is just an example, there might be a better way, so let me know what you think.

Reply:
Thanks for the suggestion @m1r0 ! I implemented it this way at first and I liked it more as well, but the issue is it doesn't probably allow us to set the default selected text to `Bulk Email Actions` instead of `Bulk Actions` as per the design. Also, we can't change the submit button's text from 'Apply' to 'Select Action'. But now that I'm thinking again and you're also suggesting, probably we can go the default `WP_List_Table`  way

https://github.com/WordPress/WordPress/blob/6d0eaf9674e550788dce7942d87af0e4f0147784/wp-admin/includes/class-wp-list-table.php#L568-L589

Reply:
Updated it here https://github.com/Automattic/sensei/pull/6552/commits/ac65f83460d99380a413ba582de58bcb0d4d83fc

Thanks again

Reply:
Nice!